### PR TITLE
Pause video when purchase form is open (mobile)

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

See title.

Refactor purchase form dialog a little bit.

## Justify why they are needed

We don't want video to be playing in the background.

Wanted to come up with a solution that doesn't cause coupling between components so went with sending custom events.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1492** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1492" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1492?utm_source=stack-comment).

File: apps/store/src/components/Video/Video.tsx [hunk 1]
```diff
@@ -75,6 +85,26 @@ export const Video = ({
     [setState, onPause],
   )
 
+  useEffect(() => {
+    const blurListener = () => {
+      if (state === State.Playing) {
+        pauseVideo()
+      }
+    }
+    window.addEventListener('dialog:open', blurListener)
```
Comment (alebedev) [inline apps/store/src/components/Video/Video.tsx:94]: Can we use enum here?

File: apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx [hunk 2]
```diff
@@ -38,13 +39,20 @@ export const PurchaseForm = () => {
   const formatter = useFormatter()
   const [{ priceIntent }, setupPriceIntent] = usePriceIntent()
   const tracking = useTracking()
+  const isLarge = useBreakpoint('lg')
 
+  const editForm = () => {
+    setFormState('EDIT')
+    if (!isLarge) {
+      sendDialogEvent('open')
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx:47]: Can we do it on dialog state change instead of here?

Comment (robinandeer) [reply apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx:47]: hmm yeah I also feel like it's awkward here but this is the event handler that opens the modal -- so in that sense it is the correct place, right?

Or do you mean we should add a `useEffect` hook inside the dialog component?

Comment (alebedev) [reply apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx:47]: I was thinking about emiting event in `onOpenChange` of dialog.  We can even make it part of our dialog wrapper in `ui` to get it working on every dialog by default

**Review by alebedev — APPROVED** [review]

File: apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/components/ProductPage/PurchaseForm/PurchaseForm.tsx:47]: Right, yes I tried that but unfortunately it's only for "managed events" like clicking the trigger, pressing escape, etc.

When the event originates from our code it doesn't get called unfortunately :P

**Comment by robinandeer** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: 6FxEzOYadpudcV0mnNmD.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1492)

**Pull request merged**